### PR TITLE
Fpi 26 merging client and json object

### PR DESCRIPTION
**Pull request by ChrisMyrants**
@broomburgo please review

### CODE REVIEW

File: Carthage/Checkouts/Abstract/RATIONALE.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Rationale
```
Comment (broomburgo) [inline Carthage/Checkouts/Abstract/RATIONALE.md]: We shouldn't commit the `Carthage/Checkouts` folder. Please remove from git and add to `.gitignore`.

**Review by broomburgo — CHANGES_REQUESTED** [review]
There's too many files to review due to the `Carthage/Checkout` commit. Please fix this before continuing the review.

File: Carthage/Checkouts/Abstract/RATIONALE.md (hunk 1 above)
Comment (ChrisMyrants) [reply Carthage/Checkouts/Abstract/RATIONALE.md]: @broomburgo ignored `Carthage/Checkouts` and deleted the folder

File: NetworkingKit/JSONObject/JSONObject.swift [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+//sourcery: prism
+public enum JSONObject {
+    case null
+    case number(JSONNumber)
+    case bool(Bool)
+    case string(String)
+    case array([JSONObject])
+    case dict([String:JSONObject])
+    
+    public static func with(_ object: Any?) -> JSONResult<JSONObject> {
```
Comment (broomburgo) [inline NetworkingKit/JSONObject/JSONObject.swift]: This should be `Any`, not `Any?`

File: NetworkingKit/JSONObject/PathTo.swift [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            return NSError.init(
+                domain: domain,
+                code: 5,
+                userInfo: ["multiple": [pathErrors.map { $0.getNSError }],
+                           NSLocalizedDescriptionKey: "PathError.multiple"])
+        }
+	}
+}
+
+extension PathError: Semigroup {
```
Comment (broomburgo) [inline NetworkingKit/JSONObject/PathTo.swift]: This could actually be made a `Monoid`. Can you think about a way to do that?

**Review by broomburgo — CHANGES_REQUESTED** [review]

**Review by broomburgo — APPROVED** [review]

**Comment by broomburgo** [comment]
Please add a minimal README, with at least the travis and codecov badges.

**Review by broomburgo — CHANGES_REQUESTED** [review]
Please add README as in comment

**Review by broomburgo — APPROVED** [review]

**Pull request merged**